lib_eio_luv/eio_luv.ml

```diff
@@ -953,6 +990,29 @@ class dir ~label (dir_path : string) = object (self)
       | exception Eio.Fs.Permission_denied (dir, ex) ->
         raise (Eio.Fs.Permission_denied (Filename.concat dir leaf, ex))
 
+  method stat path =
+    Switch.run @@ fun sw ->
+    let fd = File.open_ ~sw (self#resolve path) [`NOFOLLOW; `RDONLY] |> or_raise_path path in
+    let lst : Luv.File.Stat.t = File.fstat fd |> or_raise_path path in
+    File.close fd;
+    let mode = (Obj.magic lst.mode : int) in
```

Comment:
Is there an issue open on Luv to track this? We'll want to be notified when it's fixed.

Reply:
No, I'll open one.

Reply:
https://github.com/aantron/luv/issues/134